udp_mux.go

```diff
@@ -49,6 +50,7 @@ const maxAddrSize = 512
 type UDPMuxParams struct {
 	Logger  logging.LeveledLogger
 	UDPConn net.PacketConn
+	Net     transport.Net
```

Comment:
https://github.com/pion/ice/pull/471/files?diff=unified&w=1#diff-a8e331e8f7ba49ca6692af8fc922967307339ad6fe58eb62549602b8988d9bb3L64-L66
> 		// For unspecified addresses, the correct behavior is to return errListenUnspecified, but
> 		// it will break the applications that are already using unspecified UDP connection
> 		// with UDPMuxDefault, so print a warn log and create a local address list for mux.

Seems be used only for the fallback to the wrong usage of `NewUDPMuxDefault`.
If user want to specify `transport.New`, they should use `NewMultiUDPMuxFromPort` instead.

It would be better to add some comment to this field.

Reply:
@stv0g 
This is an API change beyond the what this PR is aiming for. I think the reason why uses vnet (or transport.Net) is just to reuse localInterfaces() that requires vnet/transport.Net. In fact, it only requires net.Interfaces, and
```go
vnet.NewNet(nil) // <-- *NetConfig being nil
````
would return what you can get from the original net.Interfaces, so the code we have here intended to use "virtual interface", and using vnet just for convenience. Therefore, I think we don't need to add transport.Net field to this public struct. (Let's focus on introducing transport.Net for now with minimal impact to users)

Reply:
I actually tried it without adding the field. However that breaks the tests as the integration will create a `UDPMux` which attempts to listen on addresses of the real host rather than the virtual interfaces of the `vnet.VNet`.

This will cause `vnet/Net.ListenUDP` to fail as it can not bind to a non-existing local address.

Reply:
For normal (non-test usage) the field will be initialized with `nil` and hence use the `stdnet.Net` implementation

Reply:
@at-wat I added a comment to the new field.